src/view/RepeatedFragment.js

```diff
@@ -210,6 +210,10 @@ export default class RepeatedFragment {
 			if ( newIndex === -1 ) return;
 
 			const fragment = this.iterations[ oldIndex ];
+
+			// guard against null fragment - #2005 (feels like treating the symptom, not the cause?)
+			if ( fragment == null ) return;
```

Comment:
Yeah, that's a little odd. Is something in the shuffle creating a null iteration? I haven't had time to look yet, but I'd guess it has something to do with the model and wrapper getting out of sync during the shuffle from the model side.